pkg/connect/client/federation/v1alpha1/federation_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2025 Cofide Limited.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"context"
+	"testing"
+
+	federationsvcpb "github.com/cofide/cofide-api-sdk/gen/go/proto/connect/federation_service/v1alpha1"
+	federationpb "github.com/cofide/cofide-api-sdk/gen/go/proto/federation/v1alpha1"
+	"github.com/cofide/cofide-api-sdk/pkg/connect/client/test"
+)
+
+const (
+	fakeFederationID      = "fake-federation-id"
+	fakeTrustZoneId       = "fake-trust-zone-id"
+	fakeRemoteTrustZoneId = "fake-remote-trust-zone-id"
+)
+
+func TestFederationClient_Unimplemented(t *testing.T) {
+	server := test.NewTestServer(t)
+	federationsvcpb.RegisterFederationServiceServer(server.Server, &federationsvcpb.UnimplementedFederationServiceServer{})
+	server.Serve()
+
+	conn := server.CreateClientConn()
+	client := New(conn)
+	ctx := context.Background()
+
+	federation, err := client.CreateFederation(ctx, nil)
+	test.RequireUnimplemented(t, err)
+	if federation != nil {
+		t.Errorf("expected nil federation, got %v", federation)
```

Comment:
minor point,no need to change - we tend to use require/assert in unit tests.

Reply:
Yeah - I guess that snuck in with some code-gen :/